src/AppBundle/Issues/IssueListener.php

```diff
@@ -58,14 +58,39 @@ public function handleCommentAddedEvent($issueNumber, $comment)
      * Adds a "Needs Review" label to new PRs.
      *
      * @param int $prNumber The number of the PR
+     * @param int $prTitle  The title of the PR
+     * @param int $prBody   The full text description of the PR
      *
      * @return string The new status
      */
-    public function handlePullRequestCreatedEvent($prNumber)
+    public function handlePullRequestCreatedEvent($prNumber, $prTitle, $prBody)
     {
+        $prLabels = array();
+
+        // new PRs always require review
         $newStatus = Status::NEEDS_REVIEW;
+        $prLabels[] = $newStatus;
+
+        // the PR title usually indicates the affected component
```

Comment:
It's more reliable, but much more complicated. I'm trying to follow the MVP philosophy:

1) Today we have "nothing".
2) This PR adds "something" (basic autolabel based on title+description).
3) In the future we'll have "everything" (smart autolabel based on modified files).

In my experience, if you try to have "everything" before "something", you'll end up with "nothing".